src/vs/workbench/browser/parts/activitybar/activitybarPart.ts

```diff
@@ -8,6 +8,7 @@
 import 'vs/css!./media/activitybarpart';
 import nls = require('vs/nls');
 import { TPromise } from 'vs/base/common/winjs.base';
+import Event, { Emitter } from 'vs/base/common/event';
```

Comment:
@octref I expect some code from here to move out to `IViewletService` or `IExtensionViewletService`. it feels lots of things that should not be activity bar related are now inside this part. To me, the part should not really distinguish between normal viewlets and extension viewlets, but just do its business of rendering a set of viewlets and their activity.